src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -85,6 +87,59 @@ public void storeMessage(Message message) {
     return messages;
   }
 
+  /**
+   * Gets messages sent between the logged-in user and another user.
+   *
+   * @return a list of messages sent between the logged-in user and another user, or empty list
+   * if logged-in user or other user have never sent a message to each other. List is sorted by time ascending.
+   */
+  public List<Message> getMessagesBetweenTwoUsers(String loggedInUser, String otherUser) {
+    List<Message> messages = new ArrayList<>();
+
+    // Messages between two people, where logged-in user is the sender
+    Filter messagesSentByLoggedInUser = new Query.FilterPredicate("user", FilterOperator.EQUAL, loggedInUser);
+    Filter messagesReceivedByOtherUser = new Query.FilterPredicate("recipient", FilterOperator.EQUAL, otherUser);
+
+    // All the messages sent by logged-in user to the other user
+    Filter loggedInUserMessages = CompositeFilterOperator.and(messagesSentByLoggedInUser, messagesReceivedByOtherUser);
+
+    // Messages between two people, where logged-in user is the recipient
+    Filter messagesSentByOtherUser = new Query.FilterPredicate("user", FilterOperator.EQUAL, otherUser);
+    Filter messagesReceivedByLoggedInUser = new Query.FilterPredicate("recipient", FilterOperator.EQUAL, loggedInUser);
+    
+    // All the messages recieved by logged-in user sent by the other user
+    Filter otherUserMessages = CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
+
+    // All the messages between the two users
+    Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+
+    Query query =
+        new Query("Message")
+            .setFilter(directMessages)
+            .addSort("timestamp", SortDirection.ASCENDING);
```

Comment:
Can you leave a comment explaining the choice of ASCENDING instead of DESCENDING? Is it because more recent messages may be in reply to previous messages?

Reply:
I added a comment, but I chose ascending order so that the most recent messages appear last/bottom of a list, like what you see in a typical thread or chat.

Reply:
I don't see the comment. Have you pushed the change? (I'm talking about a comment just before the `.addSort`, not necessarily a comment on the function as a whole.

Reply:
I haven't pushed the changes yet! I'm just replying to your comments as I make changes, so I know I've done them and then once I make all the changes, I'll push them. Sorry for the confusion.

Reply:
No worries. In the future, my recommendation would be to push changes, first, and then resolve comments. The comment resolving notifies the reviewer(s) that they should take another look.

Reply:
Got it!